portal/views/portal.py

```diff
@@ -324,6 +325,19 @@ def admin():
         u.rolelist = ', '.join([r.name for r in u.roles])
     return render_template('admin.html', users=users, wide_container="true")
 
+@portal.route('/admin/<int:user_id>/delete', methods=['GET'])
+@oauth.require_oauth()
```

Comment:
please swap the order of these two lines.  if  @roles_required() comes before @oauth.require_oauth() it'll properly redirect when a call is attempted prior to authentication.